### PR TITLE
DOC: `special.mathieu_c/sem`: add information to docstring

### LINKED ISSUE #4096: Clarify Mathieu function documentation
Hi, I came across the Mathieu functions and could not get the values from scipy.special.mathieu_cem in accordance with the solutions of the corresponding differential equation.

for q == 0 one should get simply cos(sqrt(a) \* t), but what scipy.special.mathieu_cem returns is more like cos(a \* t). So I assumed that the order parameter a is just defined differently. but then still the solutions for q != 0 do not math. wolfram alpha gives results close to the ones I got from solving the differential equation.

So I conclude the Mathieu functions in scipy are either wrong or do not correspond to the standard definition (see wikipedia, wolfram)

Cheers, Richard

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-4096

#### What does this implement/fix?
Adds requested information and examples to the documentation of `mathieu_cem` and `mathieu_sem`.

*First push with `[docs only]` in the commit message did not build the docs, so force-pushing without that...*

### CODE REVIEW

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4972]:
```suggestion
        \frac{d^2y}{dx^2} + (b_m - 2q \cos(2x))y = 0
```
Maybe do this to be suggestive of `mathieu_b`?

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4974]:
```suggestion
    for which the characteristic number :math:`b_m` results in an odd, periodic
```
I thought about explicitly mentioning that characteristic values can be found with `mathieu_b` here (and `mathieu_a` in `mathieu_cem`), but I'll wait for feedback before doing that.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4774]:
```suggestion
    solution :math:`y(x)` with period 180 degrees (for even :math:`m`) or
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4975]:
```suggestion
    solution :math:`y(x)` with period 180 degrees (for even :math:`m`) or
```

**Review by mdhaber — COMMENTED** [review]
Rendered documentation at:
https://output.circle-artifacts.com/output/job/a3bbb0f6-2528-40af-a2dd-0a97f3573d7f/artifacts/0/html/reference/special.html#mathieu-and-related-functions

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4767]:
```suggestion
    The even Mathieu functions are the solutions to Mathieu's differential equation
```

Otherwise, it's easy to start reading with emphasis on *Even*, which quickly gets confusing.
I'm not a domain expert, so please verify.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4779]: Is this valid reference format? I expect:

```
.. [1] O. McNoleg, ...
```

But perhaps SciPy follows its own standard.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4784]:
```suggestion
    Plot even Mathieu functions of orders ``2`` and ``4``.
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4798]:
```suggestion
    Because the orders ``2`` and
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [reply scipy/special/_special_ufuncs_docs.cpp:4972]: Feels like that might be confusing; we're not solving a different equation.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4985]:
```suggestion
    Plot odd Mathieu functions of orders ``2`` and ``4``.
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [inline scipy/special/_special_ufuncs_docs.cpp:4999]:
```suggestion
    Because the orders ``2`` and
```

**Review by stefanv — APPROVED** [review]
Looks good to me, @mdhaber! I made a few small stylistic suggestions, but nothing crucial.

Approving for merge, once you've made your remaining changes.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4972]: Well, it's the same form of the equation but with a different infinite set of parameters that we call $b_m$ instead of $a_m, according to the name of the function.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4779]: Is McNoleg the author of this article?

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4798]: Wait, why remove this? This is an important characteristic of the function. 
https://en.wikipedia.org/wiki/Even_and_odd_functions#:~:text=Even%20functions%20are%20those%20real,with%20respect%20to%20the%20origin.

Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4798]: Oh, I guess it would be in the first sentence then. Ok.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4779]: Oh! I messed up. I'll fix those.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4981]:
```suggestion
    .. [1] 'Mathieu function'. *Wikipedia*.
           https://en.wikipedia.org/wiki/Mathieu_function
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4780]:
```suggestion
    .. [1] 'Mathieu function'. *Wikipedia*.
           https://en.wikipedia.org/wiki/Mathieu_function
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp:4974]: As a concrete suggestion
```suggestion
    for which the characteristic number :math:`b_m`
    (which can be calculated with `mathieu_b`)
    results in an odd, periodic
```
and the change suggested [above](https://github.com/scipy/scipy/pull/21578/files#r1764484030).
(I'd make an equivalent change in the `mathieu_cem` documentation about finding the characteristic number $a_m$ with `mathieu_a`.)

Do you think I should add these changes, @stefanv? Happy to leave it as is if you don't think these changes are helpful overall. The intent is just to de-mystify these magic numbers $a_m$ that we don't provide as an argument to `mathieu_cem`/`mathieu_sem` directly and to establish the connection among some of the `mathieu`-prefixed functions.

**Review by mdhaber — COMMENTED** [review]
Thanks for the review @stefanv! Almost there, I think; just need your thoughts on whether to include the suggested references to `mathieu_a`/`mathieu_b`.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (stefanv) [reply scipy/special/_special_ufuncs_docs.cpp:4974]: Sure, why not, rather be explicit. I'd maybe just put "calculated with `mathieu_b`" in brackets to keep the sentence concise.

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4976]:
```suggestion
    for which the characteristic number :math:`b_m` (calculated with `mathieu_b`)
    results in an odd, periodic solution :math:`y(x)` with period 180 degrees
    (for even :math:`m`) or 360 degrees (for odd :math:`m`).
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [inline scipy/special/_special_ufuncs_docs.cpp:4775]:
```suggestion
    for which the characteristic number :math:`a_m` (calculated with `mathieu_a`)
    results in an odd, periodic solution :math:`y(x)` with period 180 degrees
    (for even :math:`m`) or 360 degrees (for odd :math:`m`).
```

File: scipy/special/_special_ufuncs_docs.cpp
Comment (mdhaber) [reply scipy/special/_special_ufuncs_docs.cpp]: Good idea. Done!

Comment (stefanv) [reply scipy/special/_special_ufuncs_docs.cpp]: Looks very nice, thanks!

**Pull request merged**
